src/test/java/edu/ithaca/dragon/bank/BankAccountTest.java

```diff
@@ -5,37 +5,487 @@
+
+        // missing sections -boundary cases
         assertFalse( BankAccount.isEmailValid(""));
+        assertFalse( BankAccount.isEmailValid("@"));
+        assertFalse( BankAccount.isEmailValid("."));
+        assertFalse( BankAccount.isEmailValid("@."));
+
+
+        // INVALID PREFIX
+        // invalid characters
+        assertFalse(BankAccount.isEmailValid("inv@lid123@foo.cc"));
+        assertFalse(BankAccount.isEmailValid("is.wr*ong42@mail-archive.com"));
+        assertFalse(BankAccount.isEmailValid("!nc0rre(t.domain@abc.gov"));
+        assertFalse(BankAccount.isEmailValid("n*t_g**d@foo-bar.zzzz"));
+
+        // leading '_', '.', '-'
+        assertFalse(BankAccount.isEmailValid("_leadingunderscore@test.go"));
+        assertFalse(BankAccount.isEmailValid("-this.not@good-email.com"));
+        assertFalse(BankAccount.isEmailValid(".pre.fix@doma.in"));
+        assertFalse(BankAccount.isEmailValid("_@check.it"));
+
+        // trailing '_', '.', '-'
+        assertFalse(BankAccount.isEmailValid("leadingunderscore_@test.go"));
+        assertFalse(BankAccount.isEmailValid("this.not-@good-email.com"));
+        assertFalse(BankAccount.isEmailValid("pre.fix.@doma.in"));
+        assertFalse(BankAccount.isEmailValid(".@check.it"));
+
+
+        // INVALID DOMAIN
+        // invalid characters
+        assertFalse(BankAccount.isEmailValid("domain@uh#oh.com"));
+        assertFalse(BankAccount.isEmailValid("foo-bar.baz@jun%.be"));
+        assertFalse(BankAccount.isEmailValid("abc@def^ghi.jklm"));
+
+        // leading '-'
+        assertFalse(BankAccount.isEmailValid("prefix@-abc.com"));
+        assertFalse(BankAccount.isEmailValid("prefix@-abc-def.uk"));
+        assertFalse(BankAccount.isEmailValid("prefix@-.com"));
+
+        // trailing '-'
+        assertFalse(BankAccount.isEmailValid("prefix@abc-.com"));
+        assertFalse(BankAccount.isEmailValid("prefix@abc.com-"));
+        assertFalse(BankAccount.isEmailValid("prefix@abc-def-.com"));
+        assertFalse(BankAccount.isEmailValid("prefix@abc-def.com-"));
+
+        // last part of domain length < two characters
+        assertFalse(BankAccount.isEmailValid("abc@def.a"));
+        assertFalse(BankAccount.isEmailValid("abc@def-a.b"));
     }
 
     @Test
     void constructorTest() {
         BankAccount bankAccount = new BankAccount("a@b.com", 200);
-
         assertEquals("a@b.com", bankAccount.getEmail());
         assertEquals(200, bankAccount.getBalance());
+
         //check for exception thrown correctly
         assertThrows(IllegalArgumentException.class, ()-> new BankAccount("", 100));
+
+        assertThrows(IllegalArgumentException.class, ()-> new BankAccount("a@b.com", -53));
+        assertThrows(IllegalArgumentException.class, ()-> new BankAccount("a@b.com", .125));
+        assertThrows(IllegalArgumentException.class, ()-> new BankAccount("a@b.com", 120.369));
+        assertThrows(IllegalArgumentException.class, ()-> new BankAccount("a@b.com", -3.14));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [CNT: Rough value of known constant found (CNT_ROUGH_CONSTANT_VALUE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#cnt-rough-value-of-known-constant-found-cnt-rough-constant-value).
It's recommended to use the predefined library constant for code clarity and better precision.